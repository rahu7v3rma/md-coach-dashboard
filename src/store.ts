import { configureStore } from '@reduxjs/toolkit';

import clientReducer from './reducers/client';
import groupReducer from './reducers/group';
import notificationReducer from './reducers/notification';
import userReducer from './reducers/user';

// define store options as a variable so we can export it for tests to use
const storeOptions: any = {
    reducer: {
        user: userReducer,
        client: clientReducer,
        notification: notificationReducer,
        group: groupReducer
    }
};

const store = configureStore(storeOptions);

export default store;

// infer the `RootState` type from the store
export type RootState = ReturnType<typeof store.getState>;

// infer the `AppDispatch` type from the store so thunk actions can return promises
export type AppDispatch = typeof store.dispatch;

export const test = {
    storeOptions
};
