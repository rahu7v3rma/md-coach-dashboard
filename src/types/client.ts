import { UserInfo } from './user';

export type ClientProfiles = {
    [key: string]: UserInfo;
};
