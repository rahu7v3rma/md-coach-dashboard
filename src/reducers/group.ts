import {
    createAsyncThunk,
    createEntityAdapter,
    createSlice
} from '@reduxjs/toolkit';
import { useSelector } from 'react-redux';

import * as api from '../services/api';
import { RootState } from 'src/store';
import { UserGroup } from 'src/types/user';

interface GroupState {
    loading: boolean;
    groups: UserGroup[];
}

const groupAdapter = createEntityAdapter();

const initialState = groupAdapter.getInitialState({
    loading: false
} as GroupState);

export const fetchGroupsList = createAsyncThunk(
    'chat/groups',
    async (options: {}, { rejectWithValue }) => {
        try {
            return await api.getGroupsList();
        } catch (err) {
            return rejectWithValue(err);
        }
    }
);

const groupSlice = createSlice({
    name: 'group',
    initialState,
    reducers: {},
    extraReducers: (builder) => {
        builder.addCase(fetchGroupsList.pending, (state) => {
            state.loading = true;
        });
        builder.addCase(fetchGroupsList.fulfilled, (state, action) => {
            state.loading = false;
            state.groups = action.payload;
        });
        builder.addCase(fetchGroupsList.rejected, (state) => {
            state.loading = false;
        });
    }
});

export default groupSlice.reducer;

interface GroupSelectorsType {
    loading: boolean;
    groups: UserGroup[];
}

export const GroupSelectors = (): GroupSelectorsType => {
    const loading = useSelector((state: RootState) => state.group.loading);
    const groups = useSelector((state: RootState) => state.group.groups);
    return {
        loading,
        groups
    };
};
