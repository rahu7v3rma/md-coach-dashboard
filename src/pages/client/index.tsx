import { useEffect, useState } from 'react';
import { SortOrder, TableColumn } from 'react-data-table-component';
import { useNavigate } from 'react-router-dom';
import styled from 'styled-components';
import { useDebounce } from 'use-debounce';

import ClientIcon from 'src/assets/client-icon.svg';
import { useAppDispatch } from 'src/hooks';
import {
    ClientSelectors,
    getClients,
    sortGroupOrder,
    sortOrder
} from 'src/reducers/client';
import { GroupSelectors, fetchGroupsList } from 'src/reducers/group';
import { getProfile } from 'src/reducers/user';
import { hasAuthToken } from 'src/services/auth';
import { DashboardLayout, Text } from 'src/shared';
import { Size } from 'src/shared/text';
import { UserClient, UserGroup } from 'src/types/user';
import { Colors } from 'src/utils/colors';

import DataTable from './components/datatable';
import GroupDataTable from './components/groupTable';

const TableHeader = styled.div`
    margin-top: 12px;
    margin-bottom: 24px;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
`;

const TotalCount = styled.div`
    margin-right: 12px;
    display: flex;
    flex-direction: row;
    gap: 2px;
    justify-content: space-between;
    align-items: center;
    background-color: white;
    padding: 8px 14px;
    box-shadow: 0 2px 17px rgba(0, 0, 0, 0.05);
    border-radius: 10px;
`;

const ClientImg = styled.img`
    margin-right: 8px;
`;

const Client = () => {
    const dispatch = useAppDispatch();
    const {
        clients,
        searchInput,
        totalCount,
        loading,
        sortBy,
        order,
        groupSortBy,
        groupOrder
    } = ClientSelectors();
    const { groups } = GroupSelectors();
    const navigate = useNavigate();

    const [page, setPage] = useState(1);
    const [limit, setLimit] = useState(50);
    const [value] = useDebounce(searchInput, 1500);
    useEffect(() => {
        dispatch(fetchGroupsList({}));
    }, [dispatch]);

    const handleSort = async (
        column: TableColumn<UserClient>,
        sortDirection: SortOrder
    ) => {
        dispatch(
            sortOrder({
                sortBy: column.id as number,
                order: sortDirection
            })
        );
    };

    const handleGroupSort = async (
        column: TableColumn<UserGroup>,
        sortDirection: SortOrder
    ) => {
        dispatch(
            sortGroupOrder({
                sortBy: column.id as number,
                order: sortDirection
            })
        );
    };

    useEffect(() => {
        if (!hasAuthToken()) {
            navigate('/');
        }

        dispatch(getProfile({}));
    }, [navigate, dispatch]);

    useEffect(() => {
        dispatch(
            getClients({
                page: page,
                limit: limit,
                search: value,
                sortBy: sortBy,
                order: order
            })
        );
    }, [dispatch, limit, page, value, sortBy, order]);

    return (
        <DashboardLayout>
            <GroupDataTable
                data={
                    groups?.filter((item) =>
                        item.name
                            .toLowerCase()
                            .includes(searchInput.toLowerCase())
                    ) ?? []
                }
                sortBy={groupSortBy}
                order={groupOrder}
                handleSort={handleGroupSort}
                loading={loading}
            />
            <TableHeader>
                <Text
                    color={Colors.extra.darkLiver}
                    fontSize={Size.Medium}
                    fontWeight="700"
                    lineHeight="36px"
                >
                    My Private Clients
                </Text>
                <TotalCount>
                    <ClientImg
                        alt="clients"
                        src={ClientIcon}
                        height={16}
                        width={16}
                    />
                    <Text
                        color={Colors.extra.blackText}
                        fontSize={Size.X2Small}
                        fontWeight="500"
                        lineHeight="20px"
                    >
                        {totalCount} Clients
                    </Text>
                </TotalCount>
            </TableHeader>
            <DataTable
                totalCount={totalCount}
                data={loading ? [] : clients}
                limit={limit}
                currentPage={page}
                sortBy={sortBy}
                order={order}
                pagination={true}
                onChangePage={(val) => {
                    if (!val) {
                        setPage(page - 1);
                    } else {
                        setPage(val);
                    }
                }}
                onChangeRowsPerPage={(val) => {
                    setPage(1);
                    setLimit(val);
                }}
                handleSort={handleSort}
                loading={loading}
            />
        </DashboardLayout>
    );
};

export default Client;
