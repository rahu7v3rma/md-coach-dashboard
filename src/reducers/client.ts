import {
    createAsyncThunk,
    createEntityAdapter,
    createSlice
} from '@reduxjs/toolkit';
import moment from 'moment';
import { SortOrder } from 'react-data-table-component';
import { useSelector } from 'react-redux';

import * as api from '../services/api';
import { RootState } from '../store';
import { Colors } from '../utils/colors';
import { ClientProfiles } from 'src/types/client';
import { logBooks } from 'src/types/log';
import { Note, Notes } from 'src/types/note';
import { UserClient, UserGroup } from 'src/types/user';

const clientAdapter = createEntityAdapter();

interface ClientState {
    loading: boolean;
    clients: {
        list: Array<UserClient>;
        searchInput: string;
        hasPrevious: boolean;
        hasNext: boolean;
        totalCount: number;
        nextPageNumber: number;
        previousPageNumber: number;
        sortBy: number;
        order: SortOrder;
    };
    profiles: ClientProfiles;
    logBook: logBooks;
    notes: Notes;
    groups: {
        list: Array<UserGroup>;
        searchInput: string;
        hasPrevious: boolean;
        hasNext: boolean;
        totalCount: number;
        nextPageNumber: number;
        previousPageNumber: number;
        sortBy: number;
        order: SortOrder;
    };
    analytics: any;
}

const initialState = clientAdapter.getInitialState({
    loading: false,
    profiles: {},
    logBook: {},
    notes: {},
    clients: {
        list: [],
        searchInput: '',
        hasPrevious: false,
        hasNext: false,
        totalCount: 0,
        nextPageNumber: 1,
        previousPageNumber: 0,
        sortBy: 0,
        order: 'asc'
    },
    groups: {
        list: [],
        searchInput: '',
        hasPrevious: false,
        hasNext: false,
        totalCount: 4,
        nextPageNumber: 1,
        previousPageNumber: 0,
        sortBy: 0,
        order: 'asc'
    },
    analytics: []
} as ClientState);

export const getUserInfo = createAsyncThunk(
    'client/getUserInfo',
    async (id: string, { rejectWithValue }) => {
        try {
            const userInfo = await api.getUserInfo(id);
            return {
                ...userInfo,
                id
            };
        } catch (err) {
            return rejectWithValue(err);
        }
    }
);

export const getUserAnalytics = createAsyncThunk(
    'client/getUserAnalytics',
    async (
        { id, startDate, endDate }: Record<string, string | number>,
        { rejectWithValue }
    ) => {
        try {
            return await api.getUserAnalytics(
                id as string | number,
                startDate as string,
                endDate as string
            );
        } catch (err) {
            return rejectWithValue(err);
        }
    }
);

export const getClients = createAsyncThunk(
    'client/customers',
    async (
        { page, limit, search, sortBy, order }: Record<string, string | number>,
        { rejectWithValue }
    ) => {
        try {
            return await api.getClients(
                page as number,
                limit as number,
                search as string,
                sortBy as number,
                order as SortOrder
            );
        } catch (err) {
            return rejectWithValue(err);
        }
    }
);

export const getLogBook = createAsyncThunk(
    'user/getLogBook',
    async (
        options: {
            page: number;
            limit: number;
            user_id: string;
            category: string;
        },
        { rejectWithValue }
    ) => {
        try {
            return await api.getLogBook(
                options.page,
                options.limit,
                options.user_id,
                options.category
            );
        } catch (err) {
            return rejectWithValue(err);
        }
    }
);

export const onSearchChange = createAsyncThunk(
    'onSearchChange',
    async (search: string) => {
        return search;
    }
);

export const getNotes = createAsyncThunk(
    'client/getNotes',
    async (options: { page: number; limit: number; clientId: string }) => {
        return await api.getNotes(
            options.page,
            options.limit,
            options.clientId
        );
    }
);

export const addNote = createAsyncThunk(
    'client/addNote',
    async (options: {
        id: number;
        coach_id?: string;
        client_id: string;
        content: string;
        created_at: Date;
    }): Promise<Note> => {
        return await api.addNote(+options.client_id, options.content);
    }
);

export const editNote = createAsyncThunk(
    'client/editNote',
    async (options: {
        id: number;
        coach_id?: string;
        client_id: string;
        content: string;
        created_at: Date;
    }): Promise<Note> => {
        return await api.updateNote(
            options.id,
            +options.client_id,
            options.content
        );
    }
);

export const deleteNote = createAsyncThunk(
    'client/deleteNote',
    async (options: { id: number; clientId: string }) => {
        return await api.deleteNote(options.id);
    }
);

const clientSlice: any = createSlice({
    name: 'client',
    initialState,
    reducers: {
        sortOrder: (state, action) => {
            state.clients.sortBy = action.payload.sortBy;
            state.clients.order = action.payload.order;
        },
        sortGroupOrder: (state, action) => {
            state.groups.sortBy = action.payload.sortBy;
            state.groups.order = action.payload.order;
        }
    },
    extraReducers: (builder) => {
        builder.addCase(getLogBook.pending, (state) => {
            state.loading = true;
        });
        builder.addCase(getLogBook.fulfilled, (state, action) => {
            state.loading = false;
            const userID = action.meta.arg.user_id;
            if (action.payload.has_previous) {
                const old_list = state.logBook[userID].list;
                const appended_list = [...old_list, ...action.payload.list];
                action.payload.list = appended_list;
                state.logBook[userID] = action.payload;
            } else {
                state.logBook[userID] = action.payload;
            }
        });
        builder.addCase(getLogBook.rejected, (state) => {
            state.loading = false;
        });
        builder.addCase(getUserInfo.pending, (state) => {
            state.loading = true;
        });
        builder.addCase(getUserInfo.fulfilled, (state, action) => {
            state.loading = false;

            const profiles: ClientProfiles = { ...state.profiles };
            profiles[action.payload.id] = action.payload;

            state.profiles = profiles;
        });
        builder.addCase(getUserInfo.rejected, (state) => {
            state.loading = false;
        });
        builder.addCase(getClients.pending, (state) => {
            state.loading = true;
        });
        builder.addCase(getClients.fulfilled, (state, action) => {
            state.loading = false;
            state.clients.hasPrevious = action.payload.has_previous;
            state.clients.hasNext = action.payload.has_next;
            state.clients.nextPageNumber = action.payload.next_page_number;
            state.clients.totalCount = action.payload.count;
            state.clients.list = action.payload.list;
            state.clients.sortBy = action.meta.arg.sortBy as number;
            state.clients.order = action.meta.arg.order as SortOrder;
        });
        builder.addCase(getClients.rejected, (state) => {
            state.loading = false;
        });
        builder.addCase(onSearchChange.fulfilled, (state, action) => {
            state.clients.searchInput = action.payload;
        });
        builder.addCase(getNotes.fulfilled, (state, action) => {
            const userID = action.meta.arg.clientId;
            if (action.payload.has_previous) {
                const old_list = state.notes[userID].list;
                const appended_list = [...old_list, ...action.payload.list];
                action.payload.list = appended_list;
                state.notes[userID] = action.payload;
            } else {
                state.notes[userID] = action.payload;
            }
        });
        builder.addCase(addNote.fulfilled, (state, action) => {
            state.notes[action.meta.arg.client_id].list.push(action.payload);
        });
        builder.addCase(editNote.fulfilled, (state, action) => {
            let previous_notes = {
                ...state.notes[action.meta.arg.client_id]
            };
            let edited_note = previous_notes.list.filter(
                (note) => note.id === action.meta.arg.id
            );
            if (edited_note.length) {
                edited_note[0].content = action.payload.content;
            }
            state.notes[action.meta.arg.client_id] = previous_notes;
        });
        builder.addCase(deleteNote.fulfilled, (state, action) => {
            let previous_notes = {
                ...state.notes[action.meta.arg.clientId]
            };
            previous_notes.list = previous_notes.list.filter(
                (note) => note.id !== action.meta.arg.id
            );
            state.notes[action.meta.arg.clientId] = previous_notes;
        });
        builder.addCase(getUserAnalytics.pending, (state) => {
            state.loading = true;
        });
        builder.addCase(getUserAnalytics.rejected, (state) => {
            state.loading = false;
        });
        builder.addCase(getUserAnalytics.fulfilled, (state, action: any) => {
            const UserFastData: any =
                Object.keys(action.payload.UserFast).map((key: string) => ({
                    x: moment(key).format('DD/MM/YY'),
                    y:
                        action.payload.UserFast[key]?.length > 0
                            ? action.payload.UserFast[key][0]
                                  ?.duration_minutes ?? 0
                            : 0,
                    unit: 'min'
                })) ?? [];
            const UserGlucoseData: any =
                Object.keys(action.payload.UserGlucose).map((key: string) => ({
                    x: moment(key).format('DD/MM/YY'),
                    y:
                        action.payload.UserGlucose[key]?.length > 0
                            ? action.payload.UserGlucose[key][0]?.amount ?? 0
                            : 0,
                    unit:
                        action.payload.UserGlucose[key]?.length > 0
                            ? action.payload.UserGlucose[key][0]?.unit ?? ''
                            : ''
                })) ?? [];
            const UserWeightData: any =
                Object.keys(action.payload.UserWeight).map((key: string) => ({
                    x: moment(key).format('DD/MM/YY'),
                    y:
                        action.payload.UserWeight[key]?.length > 0
                            ? Math.round(
                                  action.payload.UserWeight[key][0]?.amount
                              ) ?? 0
                            : 0,
                    unit:
                        action.payload.UserWeight[key]?.length > 0
                            ? action.payload.UserWeight[key][0]?.unit ?? ''
                            : ''
                })) ?? [];
            const UserInsulinData: any =
                Object.keys(action.payload.UserInsulin).map((key: string) => ({
                    x: moment(key).format('DD/MM/YY'),
                    y:
                        action.payload.UserInsulin[key]?.length > 0
                            ? action.payload.UserInsulin[key][0]?.units ?? 0
                            : 0,
                    unit:
                        action.payload.UserInsulin[key]?.length > 0
                            ? action.payload.UserInsulin[key][0]
                                  ?.injection_type ?? ''
                            : ''
                })) ?? [];

            state.analytics = [
                {
                    title: 'Glucose',
                    color: Colors.extra.darkTerraCotta,
                    chart: {
                        type: 'line' as 'line',
                        data: UserGlucoseData
                    }
                },
                {
                    title: 'Weight (kg)',
                    color: Colors.extra.skyBlue2,
                    chart: {
                        type: 'line' as 'line',
                        data: UserWeightData
                    }
                },
                {
                    title: 'Fasting',
                    color: Colors.extra.lightPink,
                    chart: {
                        type: 'bar' as 'bar',
                        data: UserFastData
                    }
                },
                {
                    title: 'Insulin',
                    color: Colors.extra.lightGreen2,
                    chart: {
                        type: 'bar' as 'bar',
                        data: UserInsulinData
                    }
                }
            ];
        });
    }
});

export default clientSlice.reducer;

interface ClientSelectorsType {
    loading: boolean;
    logBook: any;
    notes: Notes;
    profiles: ClientProfiles;
    clients: Array<UserClient>;
    searchInput: string;
    hasPrevious: boolean;
    hasNext: boolean;
    totalCount: number;
    totalGroupCount: number;
    nextPageNumber: number;
    previousPageNumber: number;
    sortBy: number;
    order: string;
    groups: Array<UserGroup>;
    groupSortBy: number;
    groupOrder: string;
    analytics: any;
}

export const { sortOrder, sortGroupOrder } = clientSlice.actions;

export const ClientSelectors = (): ClientSelectorsType => {
    const loading = useSelector((state: RootState) => state.client.loading);
    const clients = useSelector(
        (state: RootState) => state.client.clients.list
    );
    const groups = useSelector((state: RootState) => state.client.groups.list);
    const hasPrevious = useSelector(
        (state: RootState) => state.client.clients.hasPrevious
    );
    const hasNext = useSelector(
        (state: RootState) => state.client.clients.hasNext
    );
    const totalGroupCount = useSelector(
        (state: RootState) => state.client.groups.totalCount
    );
    const totalCount = useSelector(
        (state: RootState) => state.client.clients.totalCount
    );
    const searchInput = useSelector(
        (state: RootState) => state.client.clients.searchInput
    );
    const nextPageNumber = useSelector(
        (state: RootState) => state.client.clients.nextPageNumber
    );
    const previousPageNumber = useSelector(
        (state: RootState) => state.client.clients.previousPageNumber
    );

    const sortBy = useSelector(
        (state: RootState) => state.client.clients.sortBy
    );

    const order = useSelector((state: RootState) => state.client.clients.order);

    const groupSortBy = useSelector(
        (state: RootState) => state.client.groups.sortBy
    );

    const groupOrder = useSelector(
        (state: RootState) => state.client.groups.order
    );

    const profiles = useSelector((state: RootState) => state.client.profiles);

    const logBook = useSelector((state: RootState) => state.client.logBook);

    const notes = useSelector((state: RootState) => state.client.notes);

    const analytics = useSelector((state: RootState) => state.client.analytics);

    return {
        loading,
        logBook,
        notes,
        profiles,
        clients,
        searchInput,
        hasPrevious,
        hasNext,
        totalCount,
        nextPageNumber,
        previousPageNumber,
        sortBy,
        order,
        groups,
        totalGroupCount,
        groupSortBy,
        groupOrder,
        analytics
    };
};
