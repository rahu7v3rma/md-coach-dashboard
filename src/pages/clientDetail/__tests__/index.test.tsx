import { Provider } from 'react-redux';
import { MemoryRouter, Navigate, Route, Routes } from 'react-router-dom';
import { ReactTestRenderer, act, create } from 'react-test-renderer';

import ClientDetail from '../../clientDetail';
import store from 'src/store';

const mockDispatch = jest.fn().mockImplementation(() => {
    return {
        unwrap: jest.fn().mockImplementation(() => {
            return Promise.resolve({});
        })
    };
});

const getLogBookMockImplementation = jest
    .fn()
    .mockImplementation(() => Promise.resolve({}));

jest.spyOn(require('src/reducers/client'), 'getLogBook').mockImplementation(
    getLogBookMockImplementation
);

jest.spyOn(
    require('src/reducers/client'),
    'ClientSelectors'
).mockImplementation(
    jest.fn().mockImplementation(() => ({
        profiles: {
            '1': {
                id: 1,
                first_name: 'test',
                last_name: 'test',
                email: 'demo@gmail.com',
                contact_number: '1234567891',
                date_of_birth: '1990-01-01',
                diabetes_type: 'type 1',
                next_lesson: {
                    lesson: { title: 'Test Title', icon: 'TestIcon' },
                    order: 1
                }
            }
        },
        logBook: {
            '1': {
                next_page_number: 2,
                previous_page_number: 0,
                has_previous: true,
                has_next: true,
                list: []
            }
        }
    }))
);

jest.mock('../../../hooks', () => ({
    useAppDispatch: () => {
        return mockDispatch;
    },
    useNotificationsPermissions: jest
        .fn()
        .mockImplementationOnce(() => {
            return {
                hasPermissions: true
            };
        })
        .mockImplementation(() => {
            return {
                hasPermissions: false
            };
        })
}));

const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate
}));

describe('ClientDetail', () => {
    let tree: ReactTestRenderer;
    beforeAll(async () => {
        await act(async () => {
            tree = create(
                <Provider store={store}>
                    <MemoryRouter initialEntries={['/client/1']}>
                        <Routes>
                            <Route
                                path="/client/:id"
                                element={<ClientDetail />}
                            />
                            <Route path="*" element={<Navigate to="/" />} />
                        </Routes>
                    </MemoryRouter>
                </Provider>
            );
        });
    });

    it('renders the client detail page properly', () => {
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('displays loader properly during the loading state', async () => {
        const scrollInfinity = tree.root.findByProps({
            'data-testid': 'infinite-scroll'
        });

        await act(async () =>
            scrollInfinity.instance.onScrollListener({
                target: {
                    scrollTop: 1000,
                    scrollHeight: 2000,
                    clientHeight: 1000
                }
            })
        );
        expect(scrollInfinity.instance.state.showLoader).toBe(true);
    });

    it('handles back button click properly', async () => {
        const backBtn = tree.root.findByProps({
            'data-testid': 'backbutton'
        });
        await act(async () => backBtn.props.onClick());
        expect(mockNavigate).toHaveBeenCalledWith('/clients', {
            replace: true
        });
    });

    it('handles infinite scroll properly', async () => {
        const scrollInfinity = tree.root.findByProps({
            'data-testid': 'infinite-scroll'
        });
        await act(async () => scrollInfinity.props.next());
        expect(getLogBookMockImplementation).toHaveBeenCalledTimes(3);
    });
});
