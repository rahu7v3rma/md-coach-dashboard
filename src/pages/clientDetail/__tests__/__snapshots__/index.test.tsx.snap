// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientDetail renders the client detail page properly 1`] = `
<div
  className="sc-iKdXkR cdCBXf"
>
  <div
    className="sc-jqmfkj dxnfn"
  >
    <div
      className="sc-jfxBVY ieajce"
    >
      <div
        className="sc-ilhFlB ktqGfN"
      >
        <img
          alt="MD Icon"
          className="sc-bTuCdP dwKZtN"
          src={Object {}}
        />
        <p
          className="sc-fNZzEh fwQNpY"
        >
          Mastering 
          <br />
          Programs
        </p>
      </div>
      <div
        className="sc-fyPSjY hBHmAo"
      >
        <div
          className="sc-hKbBkQ kjgbPi"
          onClick={[Function]}
          selected={false}
        >
          <div
            className="sc-esDySZ iugWGG"
          >
            <img
              alt="My Clients"
              src={Object {}}
            />
          </div>
          <p
            className="sc-bsoyde dxwDtV"
            selected={false}
          >
            My Clients
          </p>
        </div>
        <div
          className="sc-hKbBkQ kjgbPi"
          onClick={[Function]}
          selected={false}
        >
          <div
            className="sc-esDySZ iugWGG"
          >
            <img
              alt="Chat"
              src={Object {}}
            />
          </div>
          <p
            className="sc-bsoyde dxwDtV"
            selected={false}
          >
            Chat
          </p>
        </div>
      </div>
    </div>
    <div
      className="sc-dTHgTJ dlAqcb"
    >
      <div
        className="sc-TsgdO bfDuRx"
      >
        <div
          className="sc-iRpNkT gCKbMu"
        >
          <p
            className="sc-kxtUkE ktjuwa"
            fontWeight="600"
          >
            Logout
          </p>
          <p
            className="sc-kxtUkE jsVapv"
            color="#A4AAAF"
            fontSize={11}
            fontWeight="500"
          >
            Click below to logout the dashboard
          </p>
        </div>
        <button
          className="sc-kPTPQs dzwBew"
          height="40px"
          id="logoutBtn"
          onClick={[Function]}
        >
          <p
            className="sc-kxtUkE dUXwvX"
            fontSize={12}
            fontWeight="600"
          >
            Logout
          </p>
        </button>
      </div>
    </div>
  </div>
  <div
    className="sc-fXMkjo ebnBbc"
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <div
      className="sc-dzpDMA lnJyhZ"
    >
      <div
        className="sc-eavHNl fMpyii"
      >
        <div
          className="sc-fmJwAE hLwWab"
        >
          <img
            className="sc-bHoguv fEWYAf"
            src={Object {}}
          />
          <input
            className="sc-kkscRO hZZEqs"
            onChange={[Function]}
            placeholder="Search"
          />
        </div>
      </div>
      <div
        className="sc-vdgyJ eomBSg"
      >
        <div
          className="sc-bodULI cNRoFk"
          onClick={[Function]}
        >
          <img
            className="sc-dDzzvS hlxLjW"
            src={Object {}}
          />
        </div>
        <div
          className="sc-fFGjHI fiLKAe"
        >
          <label>
            <input
              accept="image/png, image/jpeg"
              aria-label="File input"
              className="sc-gxYJeL fcIGhX"
              id="handleSelectedFileChanged"
              onChange={[Function]}
              type="file"
            />
            <div
              className="sc-dicizt hLxdUU sc-ezrZMB lgnPmK"
              id="editOverlay"
            >
              <img
                alt="avatar"
                className="sc-ewnqHT sc-ezrZMB lgnPmK"
                id="defaultAvatar"
                src={Object {}}
              />
              <img
                alt="edit"
                className="sc-iVCKna dSjaUI"
                src={Object {}}
              />
            </div>
          </label>
        </div>
      </div>
      <div
        className="sc-gLkRiC gHVgEg"
        hidden={true}
        onScroll={[Function]}
      >
        <div
          className="sc-dGYtUL goWHdG"
        >
          <span
            className="sc-fQUUyf jxQPeB"
          >
            Notifications
          </span>
          <img
            alt="Bell Icon"
            className="sc-deDhwC bzjFrc"
            onClick={[Function]}
            role="button"
            src={Object {}}
          />
        </div>
      </div>
    </div>
    <div
      className="sc-csqUdf CZBoD"
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <div
        className="sc-derkYz bnuQIs"
      >
        <div
          className="sc-bnRYnF hElrsx"
        >
          <div
            className="sc-XMxGW dIAfPc"
          >
            <img
              className="sc-ffyeCz dwDuzp"
              data-testid="backbutton"
              onClick={[Function]}
              src={Object {}}
            />
            <div
              className="sc-hYdjyz cwTWCS"
            >
              <div
                className="sc-fFGjHI fiLKAe"
              >
                <div
                  className="sc-dicizt fDkbuM sc-fkuNvB iYjqCo"
                >
                  <img
                    alt="avatar"
                    className="sc-ewnqHT sc-fkuNvB iYjqCo"
                    id="defaultAvatar"
                    src={Object {}}
                  />
                </div>
              </div>
              <div
                className="sc-flLwwW lgUxzz"
              >
                <p
                  className="sc-kxtUkE dVqPq sc-ghDWNF bUrKjz"
                >
                  test test
                </p>
                <button
                  className="sc-kPTPQs jWIHQp sc-haUagv dhFdDo"
                  height="40px"
                  onClick={[Function]}
                  width="97px"
                >
                  <img
                    className="sc-eoCneS cLOhRH"
                    src={Object {}}
                  />
                  <p
                    className="sc-kxtUkE dVqPq sc-dvmjMc ccluWs"
                  >
                    Chat
                  </p>
                </button>
              </div>
            </div>
          </div>
          <div
            className="sc-fDkjNg kJcsAJ"
          >
            <p
              className="sc-kxtUkE dVqPq sc-eGbavc UGZzn"
            >
              Next Lesson
            </p>
            <div
              className="sc-ispOId eJtfGw"
              height="100px"
              width="335px"
            >
              <img
                alt="icon"
                className="sc-cydatY iUjuvk"
                data-testid="lesson_image"
                onError={[Function]}
                src="TestIcon"
              />
              <div
                className="sc-hakzzD bJKYjU"
              >
                <p
                  className="sc-kxtUkE SwUFv"
                  color="#271A51"
                  data-testid="lesson_title"
                  fontWeight="700"
                >
                  Test Title
                </p>
                <p
                  className="sc-kxtUkE bBnJvm"
                  color="#A4AAAF"
                  data-testid="lesson_subtitle"
                  fontSize={12}
                  fontWeight="300"
                >
                  Part 1
                </p>
              </div>
              <div
                className="sc-enSMya eQGKaW"
              >
                <p
                  className="sc-kxtUkE bFrTyG"
                  color="#72996B"
                  fontSize={12}
                />
              </div>
            </div>
          </div>
          <div
            className="sc-hpzAOI fzJPXt"
          >
            <div
              className="sc-eksrvA ccWhGN"
            >
              <div
                className="sc-bgfUhJ eLtpxz"
              >
                <img
                  className="sc-gaidzC hacvHb"
                  id="phoneIcon"
                  src={Object {}}
                />
                <p
                  className="sc-kxtUkE XQNVf"
                  color="#271A51"
                  data-testid="phone_number"
                  fontSize={16}
                  fontWeight="500"
                >
                  1234567891
                </p>
              </div>
              <div
                className="sc-bgfUhJ eLtpxz"
              >
                <img
                  className="sc-gaidzC hacvHb"
                  id="emailIcon"
                  src={Object {}}
                />
                <p
                  className="sc-kxtUkE XQNVf"
                  color="#271A51"
                  data-testid="email"
                  fontSize={16}
                  fontWeight="500"
                >
                  demo@gmail.com
                </p>
              </div>
              <div
                className="sc-bgfUhJ eLtpxz"
              >
                <button
                  className="sc-kPTPQs bRxusK"
                  color="#271A51"
                  data-testid="ageButton"
                  fontSize={14}
                  id="ageButton"
                >
                  
                                    Age: 
                                    33
                                
                </button>
                <span
                  className="sc-cVVGlL cqKDik"
                />
                <button
                  className="sc-kPTPQs croZoL"
                  color="#271A51"
                  data-testid="typeButton"
                  fontSize={14}
                  id="typeButton"
                >
                  type 1
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-kjaWZf kgExRI"
        />
        <div
          className="sc-eVmQbm dEDQLD"
        >
          <div
            className="sc-deNDqs gRTwZy"
          >
            <p
              className="sc-kxtUkE dVqPq sc-eGbavc UGZzn"
            >
              Intake Form Answers
            </p>
            <div
              className="sc-chibGv kSDqKi"
            />
          </div>
          <div
            className="sc-dYdGJi fGPxSG"
          >
            <p
              className="sc-kxtUkE dVqPq sc-eGbavc UGZzn"
            >
              Logbook
            </p>
            <div
              className="infinite-scroll-component__outerdiv"
              style={Object {}}
            >
              <div
                className="infinite-scroll-component "
                style={
                  Object {
                    "WebkitOverflowScrolling": "touch",
                    "height": 550,
                    "overflow": "auto",
                  }
                }
              >
                <h4>
                  Loading...
                </h4>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
